generated-source-files.mk

```diff
@@ -123,3 +119,17 @@ genny-map-storage-repair: install-m3x-repo
 		pkg=repair \
 		value_type=ReplicaSeriesBlocksMetadata \
 		out_dir=$(m3db_package_path)/storage/repair
+
+# Map generation rule for all generated arraypools
+.PHONY: genny-arraypool-all
+genny-arraypool-all: genny-arraypool-node-segments
+
+# arraypool generation rule for ./network/server/tchannelthrift/node/segmentsArrayPool
+.PHONY: genny-arraypool-node-segments
+genny-arraypool-node-segments: install-m3x-repo
+	cd $(m3x_package_path) && make genny-arraypool                   \
+	elem_type=*rpc.Segments rename_type_prefix=segments              \
```

Comment:
Hm has this worked for you?  When I specify types that are pointer type it barfs out for me.